### PR TITLE
KEYCLOAK-8688 LDAPSyncTest is failing in some environments

### PR DESCRIPTION
**Pull request by mposolda**

### CODE REVIEW

**Comment by mposolda** [comment]
@keycloak-ci-bot test

**Comment by mposolda** [comment]
Anyone has a chance to review? This is fix in the LDAP code related to pagination. The bug causes the LDAPSyncTest fails in some environment with jenkins and with the fix, problem shouldn't happen anymore.

File: federation/ldap/src/main/java/org/keycloak/storage/ldap/idm/query/internal/LDAPQuery.java [hunk 1]
```diff
@@ -39,16 +43,19 @@
 /**
  * Default IdentityQuery implementation.
  *
+ * LDAPQuery should be closed after use in case that pagination was used (initPagination was called)
  *
  * @author Shane Bryzak
  */
 public class LDAPQuery {
```
Comment (hmlnarik) [inline federation/ldap/src/main/java/org/keycloak/storage/ldap/idm/query/internal/LDAPQuery.java]: LDAPQuery should implement AutoCloseable and used via try-with-resources

Comment (mposolda) [reply federation/ldap/src/main/java/org/keycloak/storage/ldap/idm/query/internal/LDAPQuery.java]: @hmlnarik Thanks for the suggestion and agree it is slightly better. I've updated PR to use AutoCloseable.

**Review by hmlnarik — APPROVED** [review]
Thanks @mposolda

**Review by abstractj — APPROVED** [review]

**Review by pdrozd — APPROVED** [review]

**Pull request merged**